File: SpaceLegend/Scripts_csharp/ScriptShoot.cs
using UnityEngine;
using System.Collections;

public class scriptDisparo_csharp : MonoBehaviour
{
public float velShoot = 50.0f;
public int limShoot = 40;
	// Use this for initialization
	void Start ()
	{
	
	}
	
	// Update is called once per frame
	void Update ()
	{
	transform.Translate(Vector3.up * velShoot * Time.deltaTime);
		
	if (transform.position.z >= limShoot)
		{
			Destroy(gameObject);
		}
	}
}

File: SpaceLegend/Scripts_js/scriptShoot.js
#pragma strict
var velShoot: float = 50;
var limShoot: int = -40;
var explosion: Transform;
var game : GameObject;
var explosionSound : AudioClip;

function Update () {

	transform.Translate(Vector3.up * velShoot * Time.deltaTime);

	//Remove shoot in scene
	if (transform.position.z < limShoot)
	{ 
		Destroy(gameObject);
	}

}

//Set objects behavior when collides
function OnTriggerEnter (other : Collider) {
    if (other.gameObject.tag=="rock1")
    {
	    var reset: Vector3 = Vector3(-100, Random.Range(-10.0, 10.0), Random.Range(-120.0, -170.0));
	    other.transform.position = reset;
	    
	    //Start explosion
	    Instantiate(explosion, transform.position, transform.rotation);
	    audio.PlayClipAtPoint(explosionSound, transform.position);
	    Destroy(gameObject);
	    
	    //Add point to  GameState
	    game.transform.GetComponent (scriptGameState).AddScore();
	    
    }
}